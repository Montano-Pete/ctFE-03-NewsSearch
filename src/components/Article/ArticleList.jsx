import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => (
  <ul aria-label="articles">
    {articles.map((article) => (
      <li key={`${article.title}-${article.url}`}>
        <Article
          title={article.title}
          author={article.author}
          image={article.image}
          description={article.description}
          url={article.url}
        />
      </li>
    ))}
  </ul>
);

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default ArticleList;
