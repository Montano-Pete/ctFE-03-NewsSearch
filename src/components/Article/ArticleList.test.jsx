import React from 'react';
import { screen, render } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList', () => {
  it('displays a list of articles', () => {
    render(
      <ArticleList
        articles={[
          {
            author: 'Igor Bonifacic',
            title:
              'Tesla launches Supercharger pilot program to charge other EVs',
            description:
              'As promised earlier in the year\r\n, Tesla is expanding access to its Supercharger charger network. Starting today, the company is opening 10 locations in the Netherlands to non-Tesla EVs as part of a pilot program\r\n. Provided you live in the country and you’ve…',
            url: 'https://www.engadget.com/tesla-the-netherlands-supercharger-pilot-223842483.html',
            urlToImage:
              'https://s.yimg.com/os/creatr-uploaded-images/2021-10/0ab40fa0-3b5e-11ec-a6ff-b714ec121b30',
          },
        ]}
      />
    );

    const ul = screen.findByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();
  });
});
