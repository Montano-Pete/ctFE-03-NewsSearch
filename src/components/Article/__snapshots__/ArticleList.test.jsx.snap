// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList displays a list of articles 1`] = `Promise {}`;
